### PR TITLE
Handle workflow not found in replication

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Handle workflow not found in replication

**Why?**
Handle workflow not found in replication if the workflow passes retention and gets deleted in source cluster. We ignore this workflow and clean it up in target cluster.

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/replicationTaskExecutor.go
Comment (yiminc) [inline service/history/replicationTaskExecutor.go:174]: add comment

File: service/history/replicationTaskExecutor.go
Comment (yiminc) [inline service/history/replicationTaskExecutor.go:237]: add comment

**Review by yiminc — APPROVED** [review]

File: service/history/replicationTaskExecutor.go
Comment (yycptt) [inline service/history/replicationTaskExecutor.go:298]: Synced offline. The implementation of `DeleteWorkflowExecution` will get workflow close event (for deleting vis record), which does not exist for running workflows. 

Need some refactoring work around deleting visibility records to make this work. Blocking this PR for now.

**Review by yycptt — CHANGES_REQUESTED** [review]

File: service/history/replicationTaskExecutor.go
Comment (yux0) [reply service/history/replicationTaskExecutor.go:298]: Pending https://github.com/temporalio/temporal/pull/2636

**Review by yycptt — APPROVED** [review]

File: service/history/replicationTaskExecutor.go
Comment (yycptt) [reply service/history/replicationTaskExecutor.go:298]: 👍 LGTM once #2636 is landed.

**Review by yycptt — APPROVED** [review]

**Pull request merged**
